🐛 [cloud_firestore: ^4.11.0]  Hot restarting causes app to freeze

## Bug report

**Describe the bug**
When i add Cloud fireStor package and i try to hot restart the app freezd !
when i remove and restart again the app work fine 
### Steps to reproduce

Steps to reproduce the behavior:

1. Go to '...'
2. '...'
3. See error or incorrect behavior

### Expected behavior

A clear and concise description of what you expected to happen.

### Sample project

Providing a minimal example project which demonstrates the bug in isolation from your main App _greatly_ enhances the chance of a timely fix.
Please link to the public repository URL.

---

## Additional context

Add any other context about the problem here.

---

### Flutter doctor

Run `flutter doctor` and paste the output below:
Doctor summary (to see all details, run flutter doctor -v):
[√] Flutter (Channel stable, 3.13.8, on Microsoft Windows [Version 10.0.19045.3570], locale en-US)
[√] Windows Version (Installed version of Windows is version 10 or higher)
[√] Android toolchain - develop for Android devices (Android SDK version 33.0.0-rc4)
[√] Chrome - develop for the web
[√] Visual Studio - develop Windows apps (Visual Studio Community 2022 17.0.5)
[√] Android Studio (version 2022.1)
[√] IntelliJ IDEA Community Edition (version 2022.1)
[√] VS Code (version 1.83.1)
[√] Connected device (4 available)
[√] Network resources
<details><summary>Click To Expand</summary>

```
Doctor summary (to see all details, run flutter doctor -v):
[√] Flutter (Channel stable, 3.13.8, on Microsoft Windows [Version 10.0.19045.3570], locale en-US)
[√] Windows Version (Installed version of Windows is version 10 or higher)
[√] Android toolchain - develop for Android devices (Android SDK version 33.0.0-rc4)
[√] Chrome - develop for the web
[√] Visual Studio - develop Windows apps (Visual Studio Community 2022 17.0.5)
[√] Android Studio (version 2022.1)
[√] IntelliJ IDEA Community Edition (version 2022.1)
[√] VS Code (version 1.83.1)
[√] Connected device (4 available)
[√] Network resources

• No issues found!```
</details>

---

### Flutter dependencies

Run `flutter pub deps -- --style=compact` and paste the output below:

Dart SDK 3.1.4
Flutter SDK 3.13.8
teste_cloud_firestore 1.0.0+1

dependencies:
- cupertino_icons 1.0.6
- firebase_core 2.19.0 [firebase_core_platform_interface firebase_core_web flutter meta]
- flutter 0.0.0 [characters collection material_color_utilities meta vector_math web sky_engine]

dev dependencies:
- flutter_lints 3.0.0 [lints]
- flutter_test 0.0.0 [flutter test_api matcher path fake_async clock stack_trace vector_math async boolean_selector characters collection
 material_color_utilities meta source_span stream_channel string_scanner term_glyph web]

transitive dependencies:
- async 2.11.0 [collection meta]
- boolean_selector 2.1.1 [source_span string_scanner]
- characters 1.3.0
- clock 1.1.1
- collection 1.17.2
- fake_async 1.3.1 [clock collection]
- firebase_core_platform_interface 5.0.0 [collection flutter flutter_test meta plugin_platform_interface]
- firebase_core_web 2.8.1 [firebase_core_platform_interface flutter flutter_web_plugins js meta]
- flutter_web_plugins 0.0.0 [flutter characters collection material_color_utilities meta vector_math web]
- js 0.6.7 [meta]
- lints 3.0.0
- matcher 0.12.16 [async meta stack_trace term_glyph test_api]
- material_color_utilities 0.5.0 [collection]
- meta 1.9.1
- path 1.8.3
- plugin_platform_interface 2.1.6 [meta]
- sky_engine 0.0.99
- source_span 1.10.0 [collection path term_glyph]
- stack_trace 1.11.0 [path]
- stream_channel 2.1.1 [async]
- string_scanner 1.2.0 [source_span]
- term_glyph 1.2.1
- test_api 0.6.0 [async boolean_selector collection meta source_span stack_trace stream_channel string_scanner term_glyph]
- vector_math 2.1.4
- web 0.1.4-beta

<details><summary>Click To Expand</summary>

```

```

</details>

---

when the app freez an click back button it give me this : 
Performing hot restart...
Syncing files to device sdk gphone64 x86 64...
Restarted application in 1,839ms.
D/AndroidRuntime( 2592): Shutting down VM
E/AndroidRuntime( 2592): FATAL EXCEPTION: main
E/AndroidRuntime( 2592): Process: com.example.chemsou00.dev.pharma_time, PID: 2592
E/AndroidRuntime( 2592): java.lang.RuntimeException: Unable to destroy activity {com.example.chemsou00.dev.pharma_time/com.example.chemsou00.dev.pharma_time.MainActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'void io.flutter.plugin.common.MethodChannel.setMethodCallHandler(io.flutter.plugin.common.MethodChannel$MethodCallHandler)' on a null object reference
E/AndroidRuntime( 2592): 	at android.app.ActivityThread.performDestroyActivity(ActivityThread.java:5387)
E/AndroidRuntime( 2592): 	at android.app.ActivityThread.handleDestroyActivity(ActivityThread.java:5420)
E/AndroidRuntime( 2592): 	at android.app.servertransaction.DestroyActivityItem.execute(DestroyActivityItem.java:47)
E/AndroidRuntime( 2592): 	at android.app.servertransaction.ActivityTransactionItem.execute(ActivityTransactionItem.java:45)
E/AndroidRuntime( 2592): 	at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:176)
E/AndroidRuntime( 2592): 	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:97)
E/AndroidRuntime( 2592): 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2210)
E/AndroidRuntime( 2592): 	at android.os.Handler.dispatchMessage(Handler.java:106)
E/AndroidRuntime( 2592): 	at android.os.Looper.loopOnce(Looper.java:201)
E/AndroidRuntime( 2592): 	at android.os.Looper.loop(Looper.java:288)
E/AndroidRuntime( 2592): 	at android.app.ActivityThread.main(ActivityThread.java:7839)
E/AndroidRuntime( 2592): 	at java.lang.reflect.Method.invoke(Native Method)
E/AndroidRuntime( 2592): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
E/AndroidRuntime( 2592): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1003)
E/AndroidRuntime( 2592): Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'void io.flutter.plugin.common.MethodChannel.setMethodCallHandler(io.flutter.plugin.common.MethodChannel$MethodCallHandler)' on a null object reference
E/AndroidRuntime( 2592): 	at io.flutter.plugins.firebase.firestore.FlutterFirebaseFirestorePlugin.onDetachedFromEngine(FlutterFirebaseFirestorePlugin.java:132)
E/AndroidRuntime( 2592): 	at io.flutter.embedding.engine.FlutterEngineConnectionRegistry.remove(FlutterEngineConnectionRegistry.java:272)
E/AndroidRuntime( 2592): 	at io.flutter.embedding.engine.FlutterEngineConnectionRegistry.remove(FlutterEngineConnectionRegistry.java:282)
E/AndroidRuntime( 2592): 	at io.flutter.embedding.engine.FlutterEngineConnectionRegistry.removeAll(FlutterEngineConnectionRegistry.java:290)
E/AndroidRuntime( 2592): 	at io.flutter.embedding.engine.FlutterEngineConnectionRegistry.destroy(FlutterEngineConnectionRegistry.java:123)
E/AndroidRuntime( 2592): 	at io.flutter.embedding.engine.FlutterEngine.destroy(FlutterEngine.java:460)
E/AndroidRuntime( 2592): 	at io.flutter.embedding.android.FlutterActivityAndFragmentDelegate.onDetach(FlutterActivityAndFragmentDelegate.java:763)
E/AndroidRuntime( 2592): 	at io.flutter.embedding.android.FlutterActivity.onDestroy(FlutterActivity.java:919)
E/AndroidRuntime( 2592): 	at android.app.Activity.performDestroy(Activity.java:8316)
E/AndroidRuntime( 2592): 	at android.app.Instrumentation.callActivityOnDestroy(Instrumentation.java:1364)
E/AndroidRuntime( 2592): 	at android.app.ActivityThread.performDestroyActivity(ActivityThread.java:5374)
E/AndroidRuntime( 2592): 	... 13 more
I/Process ( 2592): Sending signal. PID: 2592 SIG: 9
Lost connection to device.

--- a/packages/cloud_firestore/cloud_firestore/android/src/main/java/io/flutter/plugins/firebase/firestore/FlutterFirebaseFirestorePlugin.java
+++ b/packages/cloud_firestore/cloud_firestore/android/src/main/java/io/flutter/plugins/firebase/firestore/FlutterFirebaseFirestorePlugin.java
@@ -199,11 +199,11 @@ public class FlutterFirebaseFirestorePlugin
               Tasks.await(firestore.terminate());
               FlutterFirebaseFirestorePlugin.destroyCachedFirebaseFirestoreInstanceForKey(
                   firestore);
-
-              removeEventListeners();
-
-              taskCompletionSource.setResult(null);
             }
+            removeEventListeners();
+
+            taskCompletionSource.setResult(null);
+
           } catch (Exception e) {
             taskCompletionSource.setException(e);
           }

